Makefile

```diff
@@ -72,92 +65,76 @@ build: DARGS ?= --build-arg FROM_REG=$(BASE_REG) \
 
-# -------- Utility and Development Targets --------
-help:
-# http://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
-	@grep -E '^[a-zA-Z0-9_%/-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
-
-base/mamba_versions: pull
-base/mamba_versions: DARGS ?=
-base/mamba_versions:
-	docker run -it --rm $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_TAG) /bin/bash -c "mamba list | cat" | tr -d '\r' > $@
 
-base/distro_versions: pull
-base/distro_versions: DARGS ?=
-base/distro_versions:
-	docker run -it --rm $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_TAG)  /bin/bash -c "apt list | cat"  | tr -d '\r' > $@
+show-tag: ARGS ?=
+show-tag: DARGS ?=
+show-tag: ## tag current private build as beta
+	@-echo $(OPE_BOOK_REG)$(OPE_BOOK_IMAGE)$(OPE_BETA_TAG)
 
-versions: base/mamba_versions base/distro_versions
 
+### DEBUG TARGETS
 
 root: ARGS ?= /bin/bash
 root: DARGS ?= -u 0
 root: ## start private version  with root shell to do admin and poke around
-	-docker run -it --rm $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_BETA_TAG) $(ARGS)
+	-docker run -it --rm $(DARGS) $(OPE_BOOK_REG)$(OPE_BOOK_IMAGE)$(OPE_BETA_TAG) $(ARGS)
 
 user: ARGS ?= /bin/bash
 user: DARGS ?=
 user: ## start private version with usershell to poke around
-	-docker run -it --rm $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_BETA_TAG) $(ARGS)
+	-docker run -it --rm $(DARGS) $(OPE_BOOK_REG)$(OPE_BOOK_IMAGE)$(OPE_BETA_TAG) $(ARGS)
+
+
+### PUBLIC USAGE TARGETS
+pull: 
+pull: ## pull most recent public version
+	docker pull $(OPE_BOOK_REG)$(OPE_BOOK_IMAGE)$(OPE_PUBLIC_TAG)
 
+run: pull
 run: ARGS ?=
 run: DARGS ?= -u $(OPE_UID):$(OPE_GID) -v "${HOST_DIR}":"${MOUNT_DIR}" -v "${SSH_AUTH_SOCK}":"${SSH_AUTH_SOCK}" -v "${SSH_AUTH_SOCK}":"${SSH_AUTH_SOCK}" -e SSH_AUTH_SOCK=${SSH_AUTH_SOCK} -p ${SSH_PORT}:22
 run: PORT ?= 8888
 run: ## start published version with jupyter lab interface
-	docker run -it --rm -p $(PORT):$(PORT) $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_TAG) $(ARGS) 
-
-
-run-beta: ARGS ?=
-run-beta: DARGS ?= -u $(OPE_UID):$(OPE_GID) -v "${HOST_DIR}":"${MOUNT_DIR}" -p ${SSH_PORT}:22
-run-beta: PORT ?= 8888
-run-beta: ## start published version with jupyter lab interface
-	docker run --rm -p $(PORT):$(PORT) $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_BETA_TAG) $(ARGS)
-
-show-run-beta: ARGS ?=
-show-run-beta: DARGS ?= -u $(OPE_UID):$(OPE_GID) -v "${HOST_DIR}":"${MOUNT_DIR}" -v "${SSH_AUTH_SOCK}":"${SSH_AUTH_SOCK}" -v "${SSH_AUTH_SOCK}":"${SSH_AUTH_SOCK}" -e SSH_AUTH_SOCK=${SSH_AUTH_SOCK} -p ${SSH_PORT}:22
-show-run-beta: PORT ?= 8888
-show-run-beta: ## start published version with jupyter lab interface
-	@-echo "docker run -it --rm -p $(PORT):$(PORT) $(DARGS) $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_TAG) $(ARGS)""
-
-show-tag: ## show current tag
-	@-echo $(OPE_REGISTRY)$(OPE_IMAGE)$(OPE_BETA_TAG)
-
+	docker run -it --rm -p $(PORT):$(PORT) $(DARGS) $(OPE_BOOK_REG)$(OPE_BOOK_IMAGE)$(OPE_PUBLIC_TAG) $(ARGS) 
```

Comment:
Can you add an empty new line at the end of file?

Reply:
sure, do u mean a physical new line or \n?